e3nn/math/normalize_activation.py

```diff
@@ -1,21 +1,27 @@
 import torch
 
+from e3nn.util import torch_default_tensor_type
 
-def moment(f, n):
+
+def moment(f, n, dtype=torch.float64, device=None):
     r"""
     compute n th moment
     <f(z)^n> for z normal
     """
-    gen = torch.Generator().manual_seed(0)
-    z = torch.randn(1_000_000, generator=gen, dtype=torch.float64)
-    return f(z).pow(n).mean().item()
+    with torch_default_tensor_type(dtype, device):
+        # torch.Generator does not follow the default tensor's device type
+        gen = torch.Generator(device=device).manual_seed(0)
+        z = torch.randn(1_000_000, generator=gen)
```

Comment:
I would like to ensure reproducibility here. Therefore I would like to impose float64 and cpu to generate `z`. Then `z` can be converted

Reply:
Fixed in jnooree/e3nn@a92d418!